src/index.js

```diff
@@ -1,32 +1,47 @@
-var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
+// se pasaron las variables a const
+const XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
+// se asigna la url la variable API
+const API = 'https://rickandmortyapi.com/api/character/'
 
-var API = 'https://rickandmortyapi.com/api/character/';
-var xhttp = new XMLHttpRequest();
+const xhttp = new XMLHttpRequest();
+// La "function fetchData" la convertimos Arrow functions y hacemos que retorne un promesa: "new Promise((resolve, reject)"
+// y el texto plano de retorno de la url la convertimos en JSON: "JSON.parse(xhttp.responseText)"
+const fetchData = url_api => {
+  return new Promise((resolve, reject) => {
+    xhttp.onreadystatechange = event => {
+      if (xhttp.readyState === xhttp.DONE) { 
+        if (xhttp.status === 200) {  // -- se ejecutó el llamado exitosamente
+          resolve(JSON.parse(xhttp.responseText)); // se convierte en Json
+        } else { // -- hubo un error en el llamado
+          reject(url_api);
```

Comment:
Te recomiendo ser mas claros en el reject, esto te permite identificar el error mas fácil. 

```suggestion
          reject(new Error("Algo salió mal."));
```